Create project.

SMHFandA opened the pull request:
This is how i'm see project structure. Please, review. Maybe have comments or suggestions. I'm add unit tests later.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pllug/pandoc-slave/2)

<!-- Reviewable:end -->

SMHFandA committed:
Create project.

SMHFandA committed:
Added unit tests project.

alexchmykhalo commented on pandoc-slave/pandoc-workflow/pandocrunner.cpp:
```diff
@@ -0,0 +1,89 @@
+
+    start(mPandocExePath, params);
+
+    waitForFinished(-1);
```
We could not use "waitForFinished", as it will block main thread that cause UI freeze. Please, rewrite it in asyncronious style: method start() will run process, then signal will notify about process finish.

alexchmykhalo commented on pandoc-slave/pandoc-workflow/pandocrunner.h:
```diff
@@ -0,0 +1,68 @@
+    int statusCode() const;
+    QString content();
+    QString error();
+    QString run(PandocFormat from, PandocFormat to, const QString &file);
```
PandocRunner should also have a method that will allow it to grab all conversion results from the pandoc stdout. This will give us a way to convert document and use conversion results immediately without creating any additional files.

alexchmykhalo commented on pandoc-slave/pandoc-workflow/pandoc-workflow.pri:
```diff
@@ -0,0 +1,7 @@
+HEADERS += \
```
I suggest to rename "pandoc-workflow.pri" to "pandoc-slave.pri". As current name is not suitable for this module.

alexchmykhalo commented on pandoc-slave/main.cpp:
```diff
@@ -0,0 +1,30 @@
+    QString pandocExePath = QDir(projFolder).filePath("../../pandoc/pandoc.exe");
+    QString inputFile = QDir(projFolder).filePath("../../test-input-files/input.md");
+
+    PandocRunner runner(pandocExePath);
```
PandocRunner should have separate setter method that will take pandocExePath. In the future, we could add changing pandoc installation path during runtime.

SMHFandA committed:
1. Rewrite PandocRunner class in asynchronous style.
2. Rename pri file from pandoc-workflow.pri to pandoc-slave.pri.
3. Added setter/getter for mPandocExePath member.
4. Added reloaded run() method for content wich represents as QByteArray
instatnce.

SMHFandA committed:
1. Added method fromFormat() wich convert PandocFormat to QString;
2. Fix bug in buildParams() method: clean up mParams field.

SMHFandA committed:
Write unit tests PandocRunner class.

Guitarheroua commented:
Reviewed 6 of 14 files at r1, 10 of 10 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pllug/pandoc-slave/2)_

<!-- Sent from Reviewable.io -->

Guitarheroua commented:
Approved

[![Approved with PullApprove](https://img.shields.io/badge/pullapprove-approved-brightgreen.svg)](https://pullapprove.com/PLLUG/pandoc-slave/pull-request/2/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=PLLUG/pandoc-slave)
